test/e2e/csi_test.go

```diff
@@ -220,4 +220,322 @@ var _ = Describe("Nutanix flavor CSI", Label("capx-feature-test", "csi"), func()
+				"csi.storage.k8s.io/fstype":                             "xfs",
+				"flashMode":                                             "DISABLED",
+				"hypervisorAttached":                                    "ENABLED",
+				"storageContainer":                                      nutanixStorageContainer,
+				"storageType":                                           "NutanixVolumes",
+				"description":                                           "CAPX e2e-test",
+			},
+		}
+		Eventually(func() error {
+			return workloadClient.Create(ctx, sc)
+		}, defaultTimeout, defaultInterval).Should(Succeed())
+
+		By("Creating CSI PVC")
+		pvcName := util.RandomString(10)
+		pvc := &corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pvcName,
+				Namespace: csiNamespaceName,
+			},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes:      []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+				StorageClassName: ptr.To(csiStorageClassName),
+				Resources: corev1.VolumeResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: resource.MustParse("1Gi"),
+					},
+				},
+			},
+		}
+
+		Eventually(func() error {
+			return workloadClient.Create(ctx, pvc)
+		}, defaultTimeout, defaultInterval).Should(Succeed())
+
+		By("Creating a pod using the PVC to ensure VG is attached to the VM")
+		pod := &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "busybox-pod",
+				Namespace: csiNamespaceName,
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:            "busybox",
+						Image:           "busybox",
+						Command:         []string{"sleep", "3600"},
+						ImagePullPolicy: "IfNotPresent",
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      "csi-volume",
+								MountPath: "/data",
+							},
+						},
+					},
+				},
+				RestartPolicy: "Always",
+				Volumes: []corev1.Volume{
+					{
+						Name: "csi-volume",
+						VolumeSource: corev1.VolumeSource{
+							PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+								ClaimName: pvcName,
+							},
+						},
+					},
+				},
+			},
+		}
+
+		Eventually(func() error {
+			return workloadClient.Create(ctx, pod)
+		}, defaultTimeout, defaultInterval).Should(Succeed())
+
+		By("Checking if the pod is running")
+		podKey := client.ObjectKeyFromObject(pod)
+		Eventually(func(g Gomega) {
+			g.Expect(workloadClient.Get(ctx, podKey, pod)).To(Succeed())
+			g.Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
+		})
+
+		By("Checking CSI PVC status is bound")
+		csiPvc := &corev1.PersistentVolumeClaim{}
+		csiPvcKey := client.ObjectKey{
+			Name:      pvcName,
+			Namespace: csiNamespaceName,
+		}
+		Eventually(func(g Gomega) {
+			g.Expect(workloadClient.Get(ctx, csiPvcKey, csiPvc)).To(Succeed())
+			g.Expect(csiPvc.Status.Phase).To(Equal(corev1.ClaimBound))
+		}, defaultTimeout, defaultInterval).Should(Succeed())
+
+		By("PASSED!")
```

Comment:
is this stmt saying passed pre-mature given we delete the cluster in AfterEach funtion?

Reply:
Fair point, removed.